tests/test_pagination.py

```diff
@@ -0,0 +1,309 @@
+from asyncio import Future
+from math import ceil
+from unittest import mock
+from typing import Sequence, Dict, Optional
+
+import pytest
+from marshmallow_jsonapi import fields
+from starlette.applications import Starlette
+from starlette.responses import Response
+from starlette.testclient import TestClient
+
+from starlette_jsonapi.resource import BaseResource
+from starlette_jsonapi.pagination import (BasePaginator, BasePageNumberPaginator,
+                                          BaseCursorPaginator, BaseOffsetPaginator)
+from starlette_jsonapi.schema import JSONAPISchema
+
+
+def test_process_query_params_called_on_init():
+    paginator = BasePaginator(data=[], request=mock.MagicMock())
+    assert paginator.process_query_params() is None
+
+    process_query_params_mock = mock.MagicMock()
+    with mock.patch.object(BasePaginator, 'process_query_params', process_query_params_mock):
```

Comment:
Try the `monkeypatch` fixture instead of `mock.patch`.
[pytest docs](https://docs.pytest.org/en/stable/monkeypatch.html)